Sources/MessagingPush/PushClickHandler.swift

```diff
@@ -0,0 +1,58 @@
+import CioInternalCommon
+import CioTracking
+import Foundation
+import UserNotifications
+
+protocol PushClickHandler: AutoMockable {
+    func pushClicked(_ push: CustomerIOParsedPushPayload)
+}
+
+// sourcery: InjectRegister = "PushClickHandler"
+class PushClickHandlerImpl: PushClickHandler {
+    private let sdkConfig: SdkConfig
+    private let deepLinkUtil: DeepLinkUtil
+    private let pushHistory: PushHistory
+    private let customerIO: CustomerIOInstance
+
+    init(sdkConfig: SdkConfig, deepLinkUtil: DeepLinkUtil, pushHistory: PushHistory, customerIO: CustomerIOInstance) {
+        self.sdkConfig = sdkConfig
+        self.deepLinkUtil = deepLinkUtil
+        self.pushHistory = pushHistory
+        self.customerIO = customerIO
+    }
+
+    func pushClicked(_ parsedPush: CustomerIOParsedPushPayload) {
```

Comment:
I have encapsulated all of the logic of when a push is clicked into this class. 

Automated tests can be written for this class. That is next on my list to do.